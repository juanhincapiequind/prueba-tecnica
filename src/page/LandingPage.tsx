import { Card, Col, Container, Row } from "react-bootstrap";
import { Accommodation } from "../models/Interfaces";
import React from "react";
import LazyLoad from "react-lazyload";

const LandingPage: React.FC<{ places: Accommodation[] }> = ({ places }) => {
  return (
    <Container>
      <Row>
        {places.map((acc: Accommodation) => (
          <Col key={acc.id} sm={12} md={6} lg={4} xl={3}>
            <LazyLoad>
              <Card>
                <Card.Img variant="top" src={acc.image_url} alt="Image" />
                <Card.Body>
                  <Card.Title>{acc.name}</Card.Title>
                  <Card.Text>
                    <strong>Type:</strong> {acc.property_type} <br />
                    <strong>Amenities:</strong> {acc.amenities.join(", ")}
                  </Card.Text>
                </Card.Body>
              </Card>
            </LazyLoad>
          </Col>
        ))}
      </Row>
    </Container>
  );
};

export default LandingPage;
