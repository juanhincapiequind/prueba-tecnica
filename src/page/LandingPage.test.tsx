import { MatcherFunction, render } from "@testing-library/react";
import LandingPage from "./LandingPage";

describe("LandingPage component", () => {
  const selectedPlaces = [
    {
      id: 1,
      name: "Beach House in Miami",
      location: "Miami, Florida",
      price_per_night: 150,
      property_type: "House",
      amenities: ["Pool", "Ocean View", "WiFi"],
      image_url: "https://example.com/image1.jpg",
    },
    {
      id: 2,
      name: "Downtown Apartment in New York City",
      location: "New York City, New York",
      price_per_night: 200,
      property_type: "Apartment",
      amenities: ["Gym", "Terrace", "WiFi"],
      image_url: "https://example.com/image2.jpg",
    },
  ];

  const beachHouseMatcher: MatcherFunction = (content: string, element: Element | null) => {
    const regex = /Beach House in Miami/;
    return regex.test(content || '');
};

const downtownApartmentMatcher: MatcherFunction = (content: string, element: Element | null) => {
    const regex = /Downtown Apartment in New York City/;
    return regex.test(content || '');
};

  it("Renders accommodation card correctly", () => {
    const { getByText } = render(<LandingPage places={selectedPlaces}/>);

    expect(getByText(beachHouseMatcher)).toBeInTheDocument();
    expect(getByText(downtownApartmentMatcher)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<LandingPage places={selectedPlaces} />);
    expect(container).toMatchSnapshot();
  });
});
