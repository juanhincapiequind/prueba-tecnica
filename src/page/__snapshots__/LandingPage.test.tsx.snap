// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-xl-3 col-lg-4 col-md-6 col-sm-12"
      >
        <div
          class="lazyload-wrapper "
        >
          <div
            class="lazyload-placeholder"
          />
        </div>
      </div>
      <div
        class="col-xl-3 col-lg-4 col-md-6 col-sm-12"
      >
        <div
          class="lazyload-wrapper "
        >
          <div
            class="lazyload-placeholder"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
